x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/analysis/analyzer/Analyzer.java

```diff
@@ -1096,51 +1093,6 @@ protected LogicalPlan rule(OrderBy ob) {
         }
     }
 
-    private static class ImplicitCasting extends AnalyzerRule<LogicalPlan> {
```

Comment:
If you want to remove this class, you should probably remove its instantiation at line 100 (`// new ImplicitCasting()`)